### PR TITLE
GEOMESA-1724 HBase Spark Provider Test

### PR DESCRIPTION
**Pull request by MutahirKazmi**
This PR is primarily a test for Anthony's work on implementing Spark in HBase.  

Signed-off-by: Mutahir <kazmi.mutahir@gmail.com>

### CODE REVIEW

File: geomesa-hbase/geomesa-hbase-spark-runtime/src/test/scala/org/locationtech/geomesa/hbase/spark/HBaseSparkProviderTest.scala [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import org.locationtech.geomesa.spark.SparkSQLTestUtils
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+import scala.collection.JavaConversions._
+
+@RunWith(classOf[JUnitRunner])
+class HBaseSparkProviderTest extends Specification with LazyLogging {
+
+  // START HBASE INSTANCE MANUALLY
```
Comment (elahrvivaz) [inline geomesa-hbase/geomesa-hbase-spark-runtime/src/test/scala/org/locationtech/geomesa/hbase/spark/HBaseSparkProviderTest.scala]: should use an embedded hbase here, or rename this to `HBaseSparkProviderIntegrationTest` and skip it during the normal build

Comment (MutahirKazmi) [inline geomesa-hbase/geomesa-hbase-spark-runtime/src/test/scala/org/locationtech/geomesa/hbase/spark/HBaseSparkProviderTest.scala]: Do I need to edit POM file for skipping during normal build?

Comment (elahrvivaz) [inline geomesa-hbase/geomesa-hbase-spark-runtime/src/test/scala/org/locationtech/geomesa/hbase/spark/HBaseSparkProviderTest.scala]: you  should be able to use the specs2 `skipped("foo")` method - there should be examples in the rest of the codebase

**Comment by elahrvivaz** [comment]
@MutahirKazmi looks like the build is failing on your test: https://travis-ci.org/locationtech/geomesa/builds/210577102
Possibly the `skipped` isn't working correctly, or is in the wrong place... you might need to make things lazy and then add `skipped`s to the individual tests like here: https://github.com/locationtech/geomesa/blob/master/geomesa-hbase/geomesa-hbase-datastore/src/test/scala/org/locationtech/geomesa/hbase/data/HBaseIntegrationTest.scala

**Comment by MutahirKazmi** [comment]
@elahrvivaz just ran a complete mvn clean install

**Comment by elahrvivaz** [comment]
looks good, once the travis build passes I'll merge

**Pull request merged**